### PR TITLE
MIGRATIONS [DA-2477] - W2SC Workflow

### PR DESCRIPTION
**Pull request by dtharpeuno**
## Resolves *[ticket DA-2477]*
https://precisionmedicineinitiative.atlassian.net/browse/DA-2477

## Description of changes/additions
GHR3 project is now underway and there needs to be a system in place to move participant samples to the CVLs and then RHP for results to be generated and returned to participants. In order to move the genomics process along, a new W pipeline needs to be created at the DRC. There are many manifests that are generated or ingested by the RDR team to ensure that the process moves forward seamlessly.

** This PR is for the W2SC, WGS Secondary Confirmation Ingestion

## Tests
- [x] unit tests

### CODE REVIEW

**Review by wangy70 — APPROVED** [review]

File: rdr_service/genomic/genomic_job_components.py [hunk 1]
```diff
@@ -236,6 +236,16 @@ def generate_file_queue_and_do_ingestion(self):
             return GenomicSubProcessResult.SUCCESS if all(results) \
                 else GenomicSubProcessResult.ERROR
 
+    @staticmethod
+    def _clean_row_keys(row):
```
Comment (j-kanuch) [inline rdr_service/genomic/genomic_job_components.py:240]: Thanks for this.

File: rdr_service/genomic/genomic_job_components.py [hunk 2]
```diff
@@ -1402,13 +1406,48 @@ def _ingest_cvl_w2_manifest(self, rows):
...
+                )
+
+                if not member:
+                    logging.warning(f'Can not find genomic member record for biobank_id: '
+                                    f'{biobank_id} and sample_id: {sample_id}, skipping...')
+                    continue
+
+                member.cvlW2scManifestJobRunID = self.job_run_id
+                if member.genomicWorkflowState != GenomicWorkflowState.CVL_W2SC:
+                    member.genomicWorkflowState = GenomicWorkflowState.CVL_W2SC
```
Comment (j-kanuch) [inline rdr_service/genomic/genomic_job_components.py:1435]: Can you update the `member.genomicWorkflowStateStr` field too?

File: rdr_service/alembic/versions/00dd4c332eae_adding_w2sc_attributes.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+def downgrade(engine_name):
+    globals()["downgrade_%s" % engine_name]()
+
+
+def upgrade_rdr():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('genomic_set_member', sa.Column('cvl_w2sc_manifest_job_run_id', sa.Integer(), nullable=True))
+    op.create_foreign_key(None, 'genomic_set_member', 'genomic_job_run', ['cvl_w2sc_manifest_job_run_id'], ['id'])
+    # ### end Alembic commands ###
```
Comment (j-kanuch) [inline rdr_service/alembic/versions/00dd4c332eae_adding_w2sc_attributes.py:30]: We are going to want to create a raw table for every manifest we ingest going forward. This one can be called: `genomic_w2sc_raw`. Maybe that should be a separate PR though.

File: rdr_service/api/genomic_cloud_tasks_api.py [hunk 4]
```diff
@@ -263,6 +263,48 @@ def post(self):
...
 
 
+class IngestCVLManifestTaskApi(BaseGenomicTaskApi):
+    """
+    Cloud Task endpoint: Ingest CVL Manifest.
+    """
+    def post(self):
+        super(IngestCVLManifestTaskApi, self).post()
+
+        cvl_map = {
```
Comment (j-kanuch) [inline rdr_service/api/genomic_cloud_tasks_api.py]: Nit: can we name this something like `cvl_manifest_map` to be more specific? The name `cvl_map` makes me think it contains the actual CVL names, like Color, Baylor, UW.

File: rdr_service/resource/main.py [hunk 5]
```diff
@@ -108,6 +108,11 @@ def _build_resource_app():
                       TASK_PREFIX + "IngestAW5ManifestTaskApi",
                       endpoint="ingest_aw5_manifest_task", methods=["POST"])
 
+    # Ingest W2SC manifest
+    _api.add_resource(genomic_cloud_tasks_api.IngestCVLManifestTaskApi,
+                      TASK_PREFIX + "IngestCVLManifestTaskApi",
+                      endpoint="ingest_w2sc_manifest_task", methods=["POST"])
```
Comment (j-kanuch) [inline rdr_service/resource/main.py]: It looks like this endpoint should be more generic, like `ingest_cvl_manifest_task`.

Comment (dtharpeuno) [reply rdr_service/resource/main.py]: Updated.

File: rdr_service/api/genomic_cloud_tasks_api.py (hunk 4 above)
Comment (dtharpeuno) [reply rdr_service/api/genomic_cloud_tasks_api.py]: Sounds good.

File: rdr_service/alembic/versions/00dd4c332eae_adding_w2sc_attributes.py (hunk 3 above)
Comment (dtharpeuno) [reply rdr_service/alembic/versions/00dd4c332eae_adding_w2sc_attributes.py:30]: Yea will make a separate PR for this. What about the manifest generations as well?

File: rdr_service/genomic/genomic_job_components.py (hunk 2 above)
Comment (dtharpeuno) [reply rdr_service/genomic/genomic_job_components.py:1435]: Updated

File: rdr_service/alembic/versions/00dd4c332eae_adding_w2sc_attributes.py (hunk 3 above)
Comment (j-kanuch) [reply rdr_service/alembic/versions/00dd4c332eae_adding_w2sc_attributes.py:30]: Yea, it's probably better to have and not need than need and not have.

**Review by j-kanuch — APPROVED** [review]
Nice! First GHR3 manifest complete!
